decidim-core/app/serializers/decidim/importers/participatory_space_components_importer.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Importers
+    # This class parses and imports all components in a ParticipatorySpace.
+    # It currently supports JSON format.
+    class ParticipatorySpaceComponentsImporter < Decidim::Importers::Importer
+      # +participatory_space+: The ParticipatorySpace to which all components
+      # will belong to.
+      def initialize(participatory_space)
+        @participatory_space = participatory_space
+      end
+
+      # Parses an exported list of components and imports them into the
+      # platform.
+      #
+      # +participatory_space+: The ParticipatorySpace to which all components
+      # will belong to.
+      # +json_text+: A json document as a String.
+      # +user+: The Decidim::User that is importing.
+      def from_json(json_text, user)
+        json = JSON.parse(json_text)
+        import(json, user)
+      end
+
+      # For each component configuration in the json,
+      # creates a new Decidim::Component with that configuration.
+      #
+      # Returns: An Array with all components created.
+      #
+      # +json+: An array of json compatible Hashes with the configuration of Decidim::Components.
+      # +user+: The Decidim::User that is importing.
+      def import(json_ary, user)
+        json_ary.collect do |serialized|
```

Comment:
I think this should be wrapped in a transaction, otherwise an error could cause to have orphan or incomplete data.